src/Formats/MySQLBlockInputStream.cpp

```diff
@@ -245,7 +258,7 @@ Block MySQLBlockInputStream::readImpl()
         }
 
         ++num_rows;
-        if (num_rows == max_block_size)
+        if (num_rows == settings->max_read_mysql_row_nums || (settings->max_read_mysql_bytes_size && read_bytes_size >= settings->max_read_mysql_bytes_size))
```

Comment:
I still think that your setting `max_read_mysql_bytes` should be disabled by default. In most cases it is enough to have `max_block_size`.

Reply:
No. In most case, users' MySQL instances have a big size of one row but only use a low memory. In MySQL has `innodb_buffer_pool_size` so that it's hard to be OOM. But if use MaterializeMySQL, they will use ClickHouse as use MySQL and then the ClickHouse will be OOM when copy MySQL history data.

So we must need to control the bytes of MaterializeMySQL reads from MySQL. So we need to give the settings of `max_read_mysql_bytes` a suitable default value. Let the user do not care about how to config the ClickHouse, just enjoy it.